src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -171,12 +183,45 @@ public User getUserObject() {
         return this.user;
     }
 
+    @Inject(method = "handleClientSettings", at = @At("HEAD"))
+    public void processClientSettingsEvent(C15PacketClientSettings packet, CallbackInfo ci) {
+        PlayerChangeClientSettingsEvent event = SpongeEventFactory.createPlayerChangeClientSettingsEvent(Cause.of(NamedCause.source(this)),
```

Comment:
The event is supposed to contain the new values, but you're creating it with all of the old ones.